R/validate-config-utils.R

```diff
@@ -478,6 +478,56 @@ validate_mt_property_unique_vals <- function(model_task_grp,
   }
 }
 
+# Check that modeling task round ids match the expected round ID patterns when
+# round_id_from_variable = TRUE
+validate_mt_round_id_pattern <- function(model_task_grp,
+                                         model_task_i,
+                                         round_i,
+                                         schema,
+                                         round_id_from_variable,
+                                         round_id_var) {
+  if (!round_id_from_variable) {
+    return(NULL)
+  }
+  round_id_var_vals <- purrr::pluck(
+    model_task_grp, "task_ids", round_id_var
+  )
+  invalid_vals <- invalid_round_id_var_patterns(round_id_var_vals)
+
+  if (any(lengths(invalid_vals) > 0L)) {
+    # Collapse invalid values into a single string
+    invalid_vals_msg <- purrr::compact(invalid_vals) |>
+      purrr::map_chr(
+        ~ glue::glue_collapse(glue::glue("'{.x}'"), ", ", last = " and ")
+      )
+
+    error_row <- tibble::tibble(
+      instancePath = paste(
+        glue::glue(
+          get_error_path(schema, "/task_ids", "instance")
+        ),
+        round_id_var, names(invalid_vals_msg),
```

Comment:
Here we're creating a row for each property (`required` or `optional`) that contains invalid round_id values